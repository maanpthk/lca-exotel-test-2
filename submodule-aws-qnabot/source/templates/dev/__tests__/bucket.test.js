/*********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/                                                                               *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

const mockConfigFull = require('./mockConfigFull.json');
const mockMaster = require('./mockMaster');

function create(filename) {
    const file = `../${filename}`;
    return require(file);
}

describe('dev templates', () => {
    beforeEach(() => {
        jest.mock('../../../bin/exports', () => jest.fn(() => Promise.resolve({
            output: {
                Bucket: 'bucket',
                Prefix: 'prefix',
                devEmail: 'email',
            },
        })));
        jest.mock('../../master', () => mockMaster);
        jest.mock('../../../config.json', () => mockConfigFull);
    });

    it('renders bucket template correctly', async () => {
        const templateFile = await create('bucket');
        expect(templateFile).toMatchSnapshot({});
    });

    afterEach(() => {
        jest.resetModules();
    });
});
