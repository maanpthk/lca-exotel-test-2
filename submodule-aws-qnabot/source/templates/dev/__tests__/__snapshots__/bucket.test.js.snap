// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dev templates renders bucket template correctly 1`] = `
{
  "Description": "This template creates dev OpenSearch Cluster",
  "Outputs": {
    "Bucket": {
      "Value": {
        "Ref": "Bucket",
      },
    },
  },
  "Resources": {
    "Bucket": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "devBucketAccessLogs",
        "devBucketAccessLogsPolicy",
      ],
      "Metadata": {
        "guard": {
          "SuppressedRules": [
            "S3_BUCKET_NO_PUBLIC_RW_ACL",
          ],
        },
      },
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "LoggingConfiguration": {
          "DestinationBucketName": {
            "Ref": "devBucketAccessLogs",
          },
          "LogFilePrefix": {
            "Fn::Join": [
              "",
              [
                {
                  "Ref": "devBucketAccessLogs",
                },
                "/",
              ],
            ],
          },
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "VersioningConfiguration": {
          "Status": "Enabled",
        },
      },
      "Type": "AWS::S3::Bucket",
    },
    "CFNLambda": {
      "Metadata": {
        "cfn_nag": {
          "rules_to_suppress": [
            {
              "id": "W92",
              "reason": "This lambda function does not require to have ReservedConcurrentExecutions",
            },
          ],
        },
        "guard": {
          "SuppressedRules": [
            "LAMBDA_CONCURRENCY_CHECK",
            "LAMBDA_INSIDE_VPC",
          ],
        },
      },
      "Properties": {
        "Code": {
          "S3Bucket": undefined,
          "S3Key": {
            "Fn::Join": [
              "",
              [
                undefined,
                "/lambda/cfn.zip",
              ],
            ],
          },
        },
        "Environment": {
          "Variables": {
            "SOLUTION_ID": "SO0189",
            "SOLUTION_VERSION": "vx.x.x",
          },
        },
        "Handler": "index.handler",
        "MemorySize": "128",
        "Role": {
          "Fn::GetAtt": [
            "CFNLambdaRole",
            "Arn",
          ],
        },
        "Runtime": "nodejs",
        "Timeout": 60,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CFNLambdaPolicy": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:ListBucketVersions",
                "s3:PutBucketNotification",
                "s3:PutObject",
                "s3:GetObject",
                "s3:DeleteObjectVersion",
                "s3:DeleteObject",
                "s3:GetObjectVersion",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Sub": "arn:aws:s3:::\${Bucket}*",
                },
              ],
              "Sid": "CFNLambdaS3Access",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::ManagedPolicy",
    },
    "CFNLambdaRole": {
      "Metadata": {
        "cfn_nag": {
          "rules_to_suppress": [
            {
              "id": "W11",
              "reason": "This IAM role requires to have * resource on its permission policy",
            },
            {
              "id": "F3",
              "reason": "This role policy is required to have * action in its policy",
            },
          ],
        },
        "guard": {
          "SuppressedRules": [
            "IAM_NO_INLINE_POLICY_CHECK",
          ],
        },
      },
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Ref": "CFNLambdaPolicy",
          },
        ],
        "Path": "/",
        "Policies": [
          {
            "PolicyDocument": {
              "Statement": [
                {
                  "Action": [
                    "logs:CreateLogGroup",
                    "logs:CreateLogStream",
                    "logs:PutLogEvents",
                  ],
                  "Effect": "Allow",
                  "Resource": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:",
                        {
                          "Ref": "AWS::Partition",
                        },
                        ":logs:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        {
                          "Ref": "AWS::AccountId",
                        },
                        ":log-group:/aws/lambda/*",
                      ],
                    ],
                  },
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "LambdaFunctionServiceRolePolicy",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "Clean": {
      "DependsOn": [
        "CFNLambdaPolicy",
      ],
      "Properties": {
        "Bucket": {
          "Ref": "Bucket",
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "S3Clean",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3Clean",
    },
    "S3Clean": {
      "Metadata": {
        "guard": {
          "SuppressedRules": [
            "LAMBDA_CONCURRENCY_CHECK",
            "LAMBDA_INSIDE_VPC",
          ],
        },
      },
      "Properties": {
        "Code": {
          "S3Bucket": undefined,
          "S3Key": {
            "Fn::Join": [
              "",
              [
                undefined,
                "/lambda/s3-clean.zip",
              ],
            ],
          },
        },
        "Description": "This function clears all S3 objects from the bucket of a given S3-based resource",
        "Environment": {
          "Variables": {
            "SOLUTION_ID": "SO0189",
            "SOLUTION_VERSION": "vx.x.x",
          },
        },
        "Handler": "lambda_function.handler",
        "Role": {
          "Fn::GetAtt": [
            "CFNLambdaRole",
            "Arn",
          ],
        },
        "Runtime": "python",
        "Tags": [
          {
            "Key": "Type",
            "Value": "S3 Clean",
          },
        ],
        "Timeout": 300,
      },
      "Type": "AWS::Lambda::Function",
    },
    "devBucketAccessLogs": {
      "DeletionPolicy": "Retain",
      "Metadata": {
        "cfn_nag": {
          "rules_to_suppress": [
            {
              "id": "W35",
              "reason": "Access logging is not required for this Bucket.",
            },
          ],
        },
        "guard": {
          "SuppressedRules": [
            "S3_BUCKET_NO_PUBLIC_RW_ACL",
          ],
        },
      },
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "VersioningConfiguration": {
          "Status": "Enabled",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "devBucketAccessLogsPolicy": {
      "DependsOn": "devBucketAccessLogs",
      "Properties": {
        "Bucket": {
          "Ref": "devBucketAccessLogs",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:PutObject",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": "arn:aws:s3:::*",
                },
                "Bool": {
                  "aws:SecureTransport": "true",
                },
                "StringEquals": {
                  "aws:SourceAccount": {
                    "Ref": "AWS::AccountId",
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "Service": "logging.s3.amazonaws.com",
              },
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "devBucketAccessLogs",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "devBucketAccessLogs",
                          "Arn",
                        ],
                      },
                    ],
                  ],
                },
              ],
              "Sid": "S3ServerAccessLogsPolicy",
            },
            {
              "Action": "*",
              "Condition": {
                "Bool": {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": "*",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "devBucketAccessLogs",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "devBucketAccessLogs",
                          "Arn",
                        ],
                      },
                    ],
                  ],
                },
              ],
              "Sid": "HttpsOnly",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
}
`;
